### PR TITLE
Laravel fixes

### PR DESCRIPTION
**Pull request by duxet**

### CODE REVIEW

File: Bridges/HttpKernel.php [hunk 1]
```diff
@@ -100,6 +100,10 @@ public function onRequest(ReactRequest $request, HttpResponse $response)
         if ($this->application instanceof TerminableInterface) {
             $this->application->terminate($syRequest, $syResponse);
         }
+        
+        if (is_a($this->application, '\Illuminate\Contracts\Http\Kernel')) {
+            $this->application->terminate($syRequest, $syResponse);
+        }
```
Comment (marcj) [inline Bridges/HttpKernel.php]: Does `\Illuminate\Contracts\Http\Kernel` not implement`TerminableInterface`?

Comment (duxet) [inline Bridges/HttpKernel.php]: The only thing that Kernel implements, is just [mentioned contract](https://github.com/laravel/framework/blob/5.2/src/Illuminate/Contracts/Http/Kernel.php), that's why this one more case is required.

**Comment by duxet** [comment]
There is still some work needed to get authorization working properly, as it's singleton that stores user object as property. I tried to reset it in the same way as cookies and sessions, but there is some problem with that.

**Comment by duxet** [comment]
Authorization is finally done - i have replaced SessionGuard class with more suitable version, that clears itself after every request. It would be cool to have such thing builtin into Laravel, but at the moment i couldn't find any better solution.

**Comment by uniring** [comment]
I've tried your implementation buy I'm getting an error at line 124 of Illuminate\/Auth\/SessionGuard.php "Fatal error: Call to a member function get() on null". I'm using Laravel 5.2 if this gives you any hint.

I'll tell you if I find more info or the solution. Feel free to ask if you need something.

**Comment by OSDDQD** [comment]
@duxet can you also add a reset headers in this PR?

**Comment by andig** [comment]
@duxet any chance to fix the open errors with this pr?

File: Bridges/HttpKernel.php (hunk 1 above)
Comment (marcj) [inline Bridges/HttpKernel.php]: Is there a reason why you don't go with `$this->application instanceof \Illuminate\Contracts\Http\Kernel`?

Comment (marcj) [inline Bridges/HttpKernel.php]: ping

**Comment by marcj** [comment]
@duxet ping. We need to close if if duxet is finishing this PR.

**Comment by andig** [comment]
Superseeded

**Pull request closed without merging**